.github/workflows/test.yml

```diff
@@ -0,0 +1,48 @@
+name: Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 5
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9', 'pypy3']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Get pip cache dir
+      id: pip-cache
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key:
+          ${{ matrix.python-version }}-v1-${{ hashFiles('**/setup.py') }}
+        restore-keys: |
+          ${{ matrix.python-version }}-v1-
```

Comment:
By the way, the reason for this `v1`:

Travis CI has a way to clear caches via their web UI.

GHA doesn't. Putting an arbitrary version number in here allows us to easily bump the version to create a new cache key, and therefore generate a new cache. Useful in case the old cache is corrupt or contains junk for whatever reason.

The old cache is auto-deleted after (iirc) 14 days of no use, as usual.

Reply:
Right, I understand the reasoning, thanks for elaborating!